extensions/ql-vscode/src/data-extensions-editor/data-extensions-editor-view.ts

```diff
@@ -49,5 +71,146 @@ export class DataExtensionsEditorView extends AbstractWebview<
+        await this.clearProgress();
+        return;
+      }
+
+      await this.showProgress({
+        message: "Loading results",
+        step: 1100,
+        maxStep: 1500,
+      });
+
+      const bqrsPath = queryResult.outputDir.bqrsPath;
+
+      const bqrsChunk = await this.getResults(bqrsPath);
+      if (!bqrsChunk) {
+        await this.clearProgress();
+        return;
+      }
+
+      await this.showProgress({
+        message: "Finalizing results",
+        step: 1450,
+        maxStep: 1500,
+      });
+
+      const externalApiUsages = decodeBqrsToExternalApiUsages(bqrsChunk);
+
+      await this.postMessage({
+        t: "setExternalApiUsages",
+        externalApiUsages,
+      });
+
+      await this.clearProgress();
+    } catch (err) {
+      void showAndLogExceptionWithTelemetry(
+        redactableError(
+          asError(err),
+        )`Failed to load external APi usages: ${getErrorMessage(err)}`,
+      );
+    }
+  }
+
+  private async runQuery(): Promise<CoreCompletedQuery | undefined> {
+    const qlpacks = await qlpackOfDatabase(this.cliServer, this.databaseItem);
+
+    const packsToSearch = [qlpacks.dbschemePack];
+    if (qlpacks.queryPack) {
+      packsToSearch.push(qlpacks.queryPack);
+    }
+
+    const suiteFile = (
+      await file({
+        postfix: ".qls",
+      })
+    ).path;
+    const suiteYaml = [];
+    for (const qlpack of packsToSearch) {
+      suiteYaml.push({
+        from: qlpack,
+        queries: ".",
+        include: {
+          id: `${this.databaseItem.language}/telemetry/fetch-external-apis`,
+        },
+      });
+    }
+    await writeFile(suiteFile, dump(suiteYaml), "utf8");
+
+    const queries = await this.cliServer.resolveQueriesInSuite(
+      suiteFile,
+      getOnDiskWorkspaceFolders(),
+    );
+
+    if (queries.length !== 1) {
+      void extLogger.log(`Expected exactly one query, got ${queries.length}`);
+      return;
+    }
+
+    const query = queries[0];
+
+    const tokenSource = new CancellationTokenSource();
+
+    const queryRun = this.queryRunner.createQueryRun(
+      this.databaseItem.databaseUri.fsPath,
+      { queryPath: query, quickEvalPosition: undefined },
+      false,
+      getOnDiskWorkspaceFolders(),
+      undefined,
+      this.queryStorageDir,
+      undefined,
+      undefined,
+    );
+
+    return queryRun.evaluate(
+      (update) => this.showProgress(update, 1500),
+      tokenSource.token,
+      new TeeLogger(this.queryRunner.logger, queryRun.outputDir.logPath),
+    );
+  }
+
+  private async getResults(bqrsPath: string) {
+    const bqrsInfo = await this.cliServer.bqrsInfo(bqrsPath);
+    if (bqrsInfo["result-sets"].length !== 1) {
+      void extLogger.log(
+        `Expected exactly one result set, got ${bqrsInfo["result-sets"].length}`,
+      );
+      return undefined;
+    }
+
+    const resultSet = bqrsInfo["result-sets"][0];
+
+    await this.showProgress({
+      message: "Decoding results",
+      step: 1200,
```

Comment:
I think it's worth a comment explaining these numbers as they seem quite arbitrary.